src/types/module.rs

```diff
@@ -38,14 +38,14 @@ impl PyModule {
     ///
     /// # fn main() -> PyResult<()> {
     /// Python::with_gil(|py| -> PyResult<()> {
-    ///     let module = PyModule::new_bound(py, "my_module")?;
+    ///     let module = PyModule::new(py, "my_module")?;
     ///
     ///     assert_eq!(module.name()?, "my_module");
     ///     Ok(())
     /// })?;
     /// # Ok(())}
     ///  ```
-    pub fn new_bound<'py>(py: Python<'py>, name: &str) -> PyResult<Bound<'py, PyModule>> {
+    pub fn new<'py>(py: Python<'py>, name: &str) -> PyResult<Bound<'py, PyModule>> {
         // Could use PyModule_NewObject, but it doesn't exist on PyPy.
         let name = CString::new(name)?;
```

Comment:
This is similar to a discussion we had recently about `PyErr::warn` (https://github.com/PyO3/pyo3/pull/4364#discussion_r1685539494). Should we move this to take a `&CStr` while renaming? 

cc @mejrs, @davidhewitt

Reply:
It looks to me like `PyModule_New` just creates a `str` object and then calls `PyModule_NewObject`.

https://github.com/python/cpython/blob/d57f8a9f76e75384ec997686c2a826b1dc3c69c4/Objects/moduleobject.c#L137-L147

So maybe we should take prior art from things like `getattr` which accept either `&str` or `&Bound<'py, PyString>`, and change the implementation to call `PyModule_NewObject`?

At the same time, this is unlikely to be called in a hot loop so I'm also ok with it being a bit inefficient and keeping it simple.

Reply:
Interesting, if we change the implementation here, it's probably fine to stick with `&str` here. But how do we deal with the PyPy case then? Or is that comment not accurate anymore?

Reply:
Looks like the comment is outdated https://github.com/PyO3/python3-dll-a/blob/e7d6d1a3297d3d1531732b9ad924cd27afcc9431/src/libpypy3-c.def#L407

Reply:
Ah, in that case I would propose to change the implementation here to use `PyModule_NewObject` and stick with the current signature for simplicity.

Reply:
I can support that 👍

Reply:
Changed the implementation to `PyModule_NewObject`